astropy/units/decorators.py

```diff
@@ -250,11 +307,22 @@ def wrapper(*func_args, **func_kwargs):
             with add_enabled_equivalencies(self.equivalencies):
                 return_ = wrapped_function(*func_args, **func_kwargs)
 
-            valid_empty = (inspect.Signature.empty, None)
+            valid_empty = (inspect.Signature.empty, None, NoneType)
             if wrapped_signature.return_annotation not in valid_empty:
-                return return_.to(wrapped_signature.return_annotation)
-            else:
-                return return_
+                ret_targets = _parse_annotation(
+                    wrapped_signature.return_annotation)
+
+                # annotated output
+                # TODO drop support for Quantity (raise an Issue)
```

Comment:
These are meant for future PRs. The first is an API change where unit-related annotations must be a Unit or string, because a Unit(Quantity) creates a unit and this is probably not a desirable behavior when used in the decorator. The second is just an extension to fix #10156

Reply:
No longer relevant. Marking as resolved.